test/e2e-tests-upgrade.sh

```diff
@@ -24,11 +24,27 @@
 # Scenario 2: install the previous release, create the pipelines and tasks, upgrade
 # to the current release, and validate whether the Tekton pipeline works.
 
+# Scenario 3: install the current pipelines, use the integration tests from the previous release(s)
+# to prevent regression. Incompatible changes made in the current release cycle are expected to fail
+# and will be skipped after failing once.
+
 source $(git rev-parse --show-toplevel)/test/e2e-common.sh
 RELEASES='https://github.com/tektoncd/pipeline/releases/latest'
 PREVIOUS_PIPELINE_VERSION=$(curl -L -s -H 'Accept: application/json' $RELEASES |
   sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
 
+# BACKWARD_INCOMPATIBLE_TESTS is a manually maintained list of e2e tests that contains known breaking changes
```

Comment:
Does this mean every time we make a breaking change, we need to open a pr to add the impacted test here?

Reply:
I'm just curious how does the process look like, and who's going to maintain this. 😂 Do we need to inform all maintainers?

Reply:
Yes! great call. (this is also tracked in https://github.com/tektoncd/pipeline/issues/6709)

Currently we have a CRON job in prow https://prow.tekton.dev/?repo=tektoncd%2Fpipeline&type=periodic. Will post in slack once we have updated the tests with this PR.